AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/MutationEvent+Extensions.swift

```diff
@@ -10,16 +10,21 @@ import Dispatch
 import AWSPluginsCore
 
 extension MutationEvent {
-
-    // Updates the head of pending mutation event queue for a given model `id`
-    // if it has a `nil` version, with syncMetadata version in `mutationSync`
-    // and saves it in the mutation event table
-    static func updatePendingMutationEventVersionIfNil(for modelId: Model.Identifier,
-                                                       mutationSync: MutationSync<AnyModel>,
-                                                       storageAdapter: StorageEngineAdapter,
-                                                       completion: @escaping DataStoreCallback<Void>) {
+    // Reconciles the version of pending mutation events in mutation event table with the
+    // API response version for the following consecutive update scenarios :
+    //  - Save, then update
```

Comment:
"Save, then immediately update" makes more sense, for each scenario there are versions involves, what does the mutation queue look like and what are the versions for each operation? what does reconcile do in this scenario?